src/plot.cpp

```diff
@@ -980,16 +1019,552 @@ RGBColor random_color(void)
+  double p1 = static_cast<double>(pixels_[1]);
+  double vert_fov_radians = horiz_fov_radians * p1 / p0;
+  double dphi = horiz_fov_radians / p0;
+  double dmu = vert_fov_radians / p1;
+
+  size_t width = pixels_[0];
+  size_t height = pixels_[1];
+  ImageData data({width, height}, not_found_);
+
+  // This array marks where the initial wireframe was drawn.
+  // We convolve it with a filter that gets adjusted with the
+  // wireframe thickness in order to thicken the lines.
+  xt::xtensor<int, 2> wireframe_initial({width, height}, 0);
+
+  /* Holds all of the track segments for the current rendered line of pixels.
+   * old_segments holds a copy of this_line_segments from the previous line.
+   * By holding both we can check if the cell/material intersection stack
+   * differs from the left or upper neighbor. This allows a robustly drawn
+   * wireframe. If only checking the left pixel (which requires substantially
+   * less memory), the wireframe tends to be spotty and be disconnected for
+   * surface edges oriented horizontally in the rendering.
+   *
+   * Note that a vector of vectors is required rather than a 2-tensor,
+   * since the stack size varies within each column.
+   */
+#ifdef _OPENMP
+  const int n_threads = omp_get_max_threads();
+#else
+  const int n_threads = 1;
+#endif
+  std::vector<std::vector<std::vector<TrackSegment>>> this_line_segments(
+    n_threads);
+  for (int t = 0; t < n_threads; ++t) {
+    this_line_segments[t].resize(pixels_[0]);
+  }
+
+  // The last thread writes to this, and the first thread reads from it.
+  std::vector<std::vector<TrackSegment>> old_segments(pixels_[0]);
+
+#pragma omp parallel
+  {
+
+#ifdef _OPENMP
+    const int n_threads = omp_get_max_threads();
+    const int tid = omp_get_thread_num();
+#else
+    int n_threads = 1;
+    int tid = 0;
+#endif
+
+    SourceSite s; // Where particle starts from (camera)
+    s.E = 1;
+    s.wgt = 1;
+    s.delayed_group = 0;
+    s.particle = ParticleType::photon; // just has to be something reasonable
+    s.parent_id = 1;
+    s.progeny_id = 2;
+    s.r = camera_position_;
+
+    Particle p;
+    s.u.x = 1.0;
+    s.u.y = 0.0;
+    s.u.z = 0.0;
+    p.from_source(&s);
+
+    int vert = tid;
+    for (int iter = 0; iter <= pixels_[1] / n_threads; iter++) {
+
+      // Save bottom line of current work chunk to compare against later
+      // I used to have this inside the below if block, but it causes a
+      // spurious line to be drawn at the bottom of the image. Not sure
+      // why, but moving it here fixes things.
+      if (tid == n_threads - 1)
+        old_segments = this_line_segments[n_threads - 1];
+
+      if (vert < pixels_[1]) {
+
+        for (int horiz = 0; horiz < pixels_[0]; ++horiz) {
+
+          // Generate the starting position/direction of the ray
+          if (orthographic_width_ == 0.0) { // perspective projection
+            double this_phi = -horiz_fov_radians / 2.0 + dphi * horiz;
```

Comment:
Is this setting up particles to move through the edge or center of a pixel? It looks like edge to me.

Reply:
Did we figure out which this is doing?

Reply:
This is through edges. I see how going through centers might be desirable, do you want that? It should be imperceptible, though.

Reply:
Eh, again, sure, I'll change it to go through centers!

Reply:
OK, new commit and regolded tests go through pixel centers. You can see a slight difference in one of the images, but it's pretty negligible.

Reply:
Thanks for making that adjustment! It's generally preferable (at least in projection plots) when model extents or surface bounding boxes are used to set plot values so that the samples aren't right on the boundary of a surface -- makes it less likely we'll run into numerical artifacts in the plots. It's nice to be consistent in that approach here as well even though, as you stated, it shouldn't be noticeable for these plots.